From 5c32b4a8e7e6d66521f125fc0cb01403b9de8e62 Mon Sep 17 00:00:00 2001
From: Chew, Chiau Ee <chiau.ee.chew@intel.com>
Date: Thu, 23 Jan 2014 05:43:17 +0800
Subject: [PATCH 18/19] spi/pxa2xx-pci: Add support for Intel BYT SPI

This is enable the PCI mode support for Intel
BYT SPI controller.

This commit is created in reference to  Mika Westerberg's
previous work.

Signed-off-by: Chew, Chiau Ee <chiau.ee.chew@intel.com>
---
 drivers/spi/spi-pxa2xx-pci.c |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/drivers/spi/spi-pxa2xx-pci.c b/drivers/spi/spi-pxa2xx-pci.c
index ea42179..4643f70 100644
--- a/drivers/spi/spi-pxa2xx-pci.c
+++ b/drivers/spi/spi-pxa2xx-pci.c
@@ -11,6 +11,7 @@
 
 enum {
 	PORT_CE4100,
+	PORT_BYT,
 };
 
 struct pxa2xx_spi_pci_config {
@@ -25,6 +26,11 @@ static struct pxa2xx_spi_pci_config spi_pci_configs[] = {
 		.num_cs =  -1,
 		.bus_num = -1,
 	},
+	[PORT_BYT] = {
+		.type = LPSS_SSP,
+		.num_cs = 1,
+		.bus_num = 0,
+	},
 };
 
 static int pxa2xx_spi_pci_probe(struct pci_dev *dev,
@@ -94,6 +100,8 @@ static void pxa2xx_spi_pci_remove(struct pci_dev *dev)
 static DEFINE_PCI_DEVICE_TABLE(pxa2xx_spi_pci_devices) = {
 	{ PCI_DEVICE(PCI_VENDOR_ID_INTEL, 0x2e6a),
 	  .driver_data =  PORT_CE4100 },
+	{ PCI_DEVICE(PCI_VENDOR_ID_INTEL, 0x0f0e),
+	  .driver_data = PORT_BYT },
 	{ },
 };
 MODULE_DEVICE_TABLE(pci, pxa2xx_spi_pci_devices);
-- 
1.7.4.4

